cmd/bosun/expr/cloudwatch.go

```diff
@@ -0,0 +1,140 @@
+package expr
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	"bosun.org/cloudwatch"
+	"bosun.org/cmd/bosun/expr/parse"
+	"bosun.org/models"
+	"bosun.org/opentsdb"
+	"github.com/MiniProfiler/go/miniprofiler"
+)
+
+// cloudwatch defines functions for use with amazon cloudwatch api
+var CloudWatch = map[string]parse.Func{
+
+	"cloudwatch": {
+		Args: []models.FuncType{models.TypeString, models.TypeString, models.TypeString, models.TypeString,
+			models.TypeString, models.TypeString, models.TypeString, models.TypeString},
+		Return: models.TypeSeriesSet,
+		Tags:   cloudwatchTagQuery,
+		F:      CloudWatchQuery,
+	},
+}
+
+func parseCloudWatchResponse(req *cloudwatch.Request, s *cloudwatch.Response) ([]*Result, error) {
+	const parseErrFmt = "cloudwatch ParseError (%s): %s"
+	if s == nil {
+		return nil, fmt.Errorf(parseErrFmt, req.Metric, "empty response")
+	}
+	results := make([]*Result, 0)
+	dps := make(Series)
+	tags := make(opentsdb.TagSet)
+	for _, d := range req.Dimensions {
+		tags[d.Name] = d.Value
+	}
+
+	switch req.Statistic {
+	case "Sum":
+		for _, x := range s.Raw.Datapoints {
+			dps[*x.Timestamp] = *x.Sum
+		}
+	case "Average":
+		for _, x := range s.Raw.Datapoints {
+			dps[*x.Timestamp] = *x.Average
+		}
+	case "Minimum":
+		for _, x := range s.Raw.Datapoints {
+			dps[*x.Timestamp] = *x.Minimum
+		}
+	case "Maximum":
+		for _, x := range s.Raw.Datapoints {
+			dps[*x.Timestamp] = *x.Maximum
+		}
+	default:
+		return nil, fmt.Errorf("No such statistic '%s'", req.Statistic)
+	}
+
+	results = append(results, &Result{
+		Value: dps,
+		Group: tags,
+	})
+	return results, nil
+}
+
+func parseDimensions(dimensions string) []cloudwatch.Dimension {
+	parsed := make([]cloudwatch.Dimension, 0)
+	dims := strings.Split(dimensions, ",")
+	for _, row := range dims {
+		dim := strings.Split(row, ":")
+		parsed = append(parsed, cloudwatch.Dimension{Name: dim[0], Value: dim[1]})
+	}
+	return parsed
+}
+
+func CloudWatchQuery(e *State, T miniprofiler.Timer, region, namespace, metric, period, statistic, dimensions, sduration, eduration string) (*Results, error) {
```

Comment:
Can you add some docs that explain how to use this please? Its a bit hard to visualize without an example as the end-user will call into this.

Reply:
Added documentation for the public method, configuration section and a full example of an alert using cloud watch metrics